### PR TITLE
feat(unlock-app) - get lockAddress for lock that use a past version of Unlock contract

### PR DESCRIPTION
**Pull request by kalidiagne**
# Description
fix issue with undefined lock address for locks with old contract address 

# Issues

Fixes #
Refs #

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [x] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: unlock-app/src/components/interface/verification/Key.tsx [hunk 1]
```diff
@@ -172,6 +172,8 @@ export const ValidKey = ({
   const storageService = useStorageService()
   const walletService = useContext(WalletServiceContext)
 
+  console.log(lock)
```
Comment (julien51) [inline unlock-app/src/components/interface/verification/Key.tsx:175]: Remove please

File: unlock-app/src/components/interface/verification/Key.tsx [hunk 2]
```diff
@@ -205,14 +207,17 @@ export const ValidKey = ({
         return
       }
       await siweLogin()
+
+      // get lockAddress for lock that use a past version of Unlock contract
+      const lockAddress = lock?.address || lock?.unlockContractAddress
```
Comment (julien51) [inline unlock-app/src/components/interface/verification/Key.tsx:212]: That is not the lock address. Unlock contract is different from lock contract!!!

**Review by julien51 — CHANGES_REQUESTED** [review]
I'm doing thing this solves anything.

File: packages/unlock-js/src/web3Service.ts [hunk 3]
```diff
@@ -89,7 +89,7 @@ export default class Web3Service extends UnlockService {
       address,
       this.providerForNetwork(network)
     )
-    const lock = version.getLock.bind(this)(
+    const lock = await version.getLock.bind(this)(
```
Comment (kalidiagne) [inline packages/unlock-js/src/web3Service.ts:92]: @julien51 looks like `await` is missing here, thats why `address` is not present.

Comment (julien51) [reply packages/unlock-js/src/web3Service.ts:92]: Very good catch

**Pull request merged**